File: Skydown App/Models/CartItem/CartItem.swift
//
//  CartItem.swift
//  Skydown App
//
//  Created by Yang D. Nash on 19.08.25.
//

import Foundation

struct CartItem: Identifiable {
    let id = UUID()
    let item: MerchandiseItem
    let size: String
    let color: String?
    var quantity: Int
    let shopifyVariantId: String?
    let sku: String?
    let unitPrice: Double?

    var effectiveUnitPrice: Double {
        unitPrice ?? item.price
    }
}

File: shared/src/commonMain/kotlin/com/skydown/shared/model/CartItem.kt
package com.skydown.shared.model

import kotlinx.serialization.Serializable

@Serializable
data class CartItem(
    val item: MerchandiseItem,
    val size: String,
    val color: String? = null,
    val quantity: Int,
    val shopifyVariantId: String? = null,
    val sku: String? = null,
    val unitPrice: Double? = null,
)
